unit_tests/test_redis_aggregator.py

```diff
@@ -0,0 +1,238 @@
+                                 "batch_commands": TEST_BATCH_COMMAND_DICT}
+        batched_channel_input = f"external//aggregator/{TEST_AGGREGATOR_UUID}/batch_commands"
+
+        aggregator.execute_batch_commands(is_blocking=False)
+        aggregator.redis_db.publish.assert_called_with(batched_channel_input,
+                                                       json.dumps(batched_command_input))
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+    def test_execute_batch_commands_api_time_out_raise_exception(aggregator):
+        with patch("gsy_e_sdk.redis_aggregator.wait_until_timeout_blocking",
+                   side_effect=AssertionError):
+            aggregator.device_uuid_list = [TEST_DEVICE_UUID_1, TEST_DEVICE_UUID_2]
+            aggregator.aggregator_uuid = TEST_AGGREGATOR_UUID
+            with pytest.raises(RedisAggregatorAPIException):
+                aggregator.execute_batch_commands()
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+    @pytest.mark.parametrize("is_blocking_, trans_id_resp_buffer, expected_ret_val",
+                             [(True, TEST_TRANSACTION_ID_RESPONSE_BUFFER, TEST_RESPONSE),
+                              (False, TEST_TRANSACTION_ID_RESPONSE_BUFFER, None),
+                              (True, {}, None)])
+    def test_execute_batch_commands_returns_expected(aggregator, is_blocking_,
+                                                     trans_id_resp_buffer, expected_ret_val):
+        aggregator.device_uuid_list = [TEST_DEVICE_UUID_1, TEST_DEVICE_UUID_2]
+        aggregator.aggregator_uuid = TEST_AGGREGATOR_UUID
+        aggregator._transaction_id_response_buffer = trans_id_resp_buffer
+
+        assert aggregator.execute_batch_commands(is_blocking=is_blocking_) is expected_ret_val
+
+
+@pytest.mark.usefixtures("mock_grid_fee_calculation")
+class TestRedisAggregatorCalculateGridFee:
+    """Test cases for RedisAggregator's calculate_grid_fee method."""
+
+    @staticmethod
+    def test_calculate_grid_fee_called_with_args(aggregator):
+        aggregator.calculate_grid_fee("start_market",
+                                      "target_market",
+                                      "current_market_fee")
+        aggregator.grid_fee_calculation.calculate_grid_fee.assert_called_with(
+            "start_market",
+            "target_market",
+            "current_market_fee")
+
+    @staticmethod
+    def test_calculate_grid_fee_returns_expected(aggregator):
+        aggregator.grid_fee_calculation.calculate_grid_fee.return_value = "TEST_OK"
+        ret_val = aggregator.calculate_grid_fee("start_market",
+                                                "target_market",
+                                                "current_market_fee")
+        assert ret_val == "TEST_OK"
+
+
+def test_get_uuid_from_area_name_returns_expected(aggregator):
```

Comment:
This is so, because there were no other tests regarding the same method in order to group them all together in one single class.
I thought I could maybe include these loose tests in a class called

```
class TestRedisAggregator<name_which_describes_loose_tests>:
    def @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking"):
        # ...
```

but I did not find any suitable name. Also, in the past PR there were no comments about an analogous case so I assumed there was no problem. I'm open to suggestions.

Reply:
This also caught my eyes, I prefer grouping all tests in a class (well I personally think that all of these tests could be grouped in a single class) however I thought that us it's okay because we know that all of the redis aggregator tests are in the same module `test_redis_aggregator`
So I'm okay with both approaches

Reply:
So what would be our agreement here @fievelk ?

Reply:
I left a similar question in [here](https://github.com/gridsingularity/gsy-e-sdk/pull/194#discussion_r775394173)
I think that grouping could be useful, but then it is also complicated, because: Have a look at the `test_constructor_grid_fee_calculation_instantiated`, IMO it could also reside here in the `TestRedisAggregatorCalculateGridFee`. 
I do not have a strong opinion here, as we do not have to search for tests very often so having one test class for all test methods works, too.

Reply:
@ERafaelMartinez I've started a conversation about this on Slack. I would say that the general agreement is this:
- tests should be defined inside of classes. Even though Pytest examples define functions in the module scope, we've decided to avoid this (you can check the conversation on slack for more information).
- there should be only one test class for each tested class. There can be exceptions like in this PR, where you created one class for each method; however, the first approach should be preferred when possible.

So class-based is fine. We would like to avoid defining test functions in the module scope. If you've decided to create 1 class for each tested module, please do that also for methods that need only one test. If this means creating too many test classes, then it might be another good reason to actually avoid this approach and just create one single test class for each tested class.

Reply:
As there would be to many classes due to this one-test-only methods, I'll address this discussion here by refactoring all test in a single class named _TestRedisAggregator_.

Reply:
Thanks for the discussion.